src/raft.rs

```diff
@@ -888,12 +1002,106 @@ impl<T: Storage> Raft<T> {
     /// according to the progress recorded in r.prs().
     pub fn bcast_append(&mut self) {
```

Comment:
follower replication only covers `bcast_append`, but misses somewhere calling `send_append` and `send_append_aggressively` directly

Reply:
When a cluster replicates data via `bcast_append`, the cluster is in a stable state with high probability.
If `send_append` is called, it is usually due to reject of `MsgAppendResp`, which means that follower's raft log is conflict with leader's. In this situation, we'd better use leader replication to fix follower's raft log as quickly as possible.

Reply:
In previous test, we find that follower replication can reduce a lot of cross-az traffic. So I think we should sacrifice some coverage of follower replication to archive less performance regression. Though we just use follower replication in the main path of log replication, it can still reduce considerable cross-az traffic.
Maybe we can expand follower replication in the future. For example, we can get snapshot from followers.